### ISSUE TITLE
Safer latex_dependency insertion

### ISSUE THREAD

**Issue by cderv** [rstudio/rmarkdown#2546]
https://github.com/rstudio/rmarkdown/blob/ee69d59f8011ad7b717a409fcbf8060d6ffc4139/R/latex_dependencies.R#L48-L59

I wonder if it would be ok to use safer inclusion to avoid potential conflict if this is used in a document where the is already loaded package 🤔 

Using `\@ifpackageloaded{pkgname}{}{\usepackage[options]{name}}` 

If a package needs options to be set though, there is not really a good way to avoid option conflict, unless this is loaded before `documentclass` with `\PassOptionsToPackage{options}{pkgname}` 

Anyhow, I feel `latex_dependency` is not that much used, and if it would be it need to be safer. 

For thoughts...

**Comment by yihui** [comment]
Perhaps using `\@ifpackageloaded{pkgname}` only when `opts == ''`? That means, if users don't need any options, we can guarantee that the package will be loaded without conflicts. If they do, we just respect the options, which may or may not lead to conflicts.

**Comment by cderv** [comment]
Yes definitely - `\@ifpackageloaded{pkgname}` should be only used if `opts == ''`. I like that. thanks !
